### PR TITLE
Migrate chart from helm/stable, add license and CI/CD pipelines

### PR DESCRIPTION
**Pull request by erkannt**
- Chart history migrated to the new repository by @JoelSpeed 
- Apache license taken from helm/stable repo
- pipelines based on [helm/charts-repo-actions-demo](https://github.com/helm/charts-repo-actions-demo)
  - same linting tool as used by helm/stable repo
  - successfully ran `ct lint` and `ct install` locally (see readme in this PR on how to do this)

If merged to the main branch this should get us a gh-pages hosted helm chart running the same validations as before.

I'm using my fork to test if the workflows work as intended. Will report if successful.

### CODE REVIEW

**Comment by JoelSpeed** [comment]
Thanks for working on this @erkannt! Would you mind rebasing your work on this branch I pushed over the weekend? https://github.com/oauth2-proxy/manifests/tree/migration

The branch I pushed retains as much history from the original chart as possible and would be preferable rather than introducing the chart in a single new commit

**Comment by erkannt** [comment]
No worries, my pleasure.

I set this PR up to merge into the branch you mention. So I think it should build on top of it, thereby keeping the history. Would you prefer me to rebase squash my commits so that we end up with a single commit on the `migration` branch?

**Comment by erkannt** [comment]
Took some fixes, but the `lint-test` and `release` workflows have successfully worked in the forked repo:
https://github.com/erkannt/manifests/actions

**Comment by erkannt** [comment]
One important piece of prep before merging into main:

the repo needs to have a `gh-pages` branch for the `chart-releaser` to do its magic. If you'd  prefer a different branch name this can be reflected by passing a flag or config file to the chart-releaser action.

**Comment by JoelSpeed** [comment]
> I set this PR up to merge into the branch you mention. So I think it should build on top of it, thereby keeping the history. Would you prefer me to rebase squash my commits so that we end up with a single commit on the migration branch?

I didn't realise that this PR was based on the migration branch! I don't mind having multiple commits in the PR provided they are atomic and logical, eg would be good to squash commits that are fixes for previous issues in the same PR.

> the repo needs to have a gh-pages branch for the chart-releaser to do its magic. If you'd prefer a different branch name this can be reflected by passing a flag or config file to the chart-releaser action.

Does this need to be created or can the releaser create the branch? I think using the gh-pages branch should be fine for now, I assume we could change it later if need be

---

I noticed you move the chart from `chart/` to `charts/oauth2-proxy`, this seems like it's probably idiomatic for a project that is going to host multiple charts, but I suspect we aren't ever going to need to do that? Does it still make sense to have the chart nested in that extra level?

**Comment by erkannt** [comment]
- the branch has to be created by a human prior to running the releaser
- I'll squash the fixes this evening (UK time)
- and I'll move the chart folder back to how it was before (YAGNI and such); I initially did it due to chart-test being quite opinionated, but one can configure it to other paths

**Comment by erkannt** [comment]
- collapsed the fix commits
- didn't rename the `chart` folder
- only bumped chart by one patch increment

Force pushed to update this PR.

/cc @desaintmartin @tlawrie as you have been added to the `maintainers` in the `Chart.yaml`

I did this as you were mentioned in the thread that led to this repo being set up. Hope that is as intended.

File: chart/pdb-values.yaml
Comment (desaintmartin) [inline chart/pdb-values.yaml:2]: Should we keep those in the ci directory, along with a blank default-values.yaml?

**Comment by desaintmartin** [comment]
Wow, great work, thanks!
I suppose next step is to merge and reference it in the artifact hub.
BTW, before merging, it may help having their `artifacthub-repo.yml` file. I can do in a follow-up PR if needed.

**Comment by erkannt** [comment]
No worries. Happy to help where I can.

I'm not sure what should go in the `artifact-repo.yml` so it would be great if you could do that PR.

Prior to merging a `gh-pages` branch needs to be created in the repo (by a human) so that the publisher action can use it.

**Comment by ekeih** [comment]
Hi, thanks for working on a new official chart.
I think we should replace the image to the new official one (`quay.io/oauth2-proxy/oauth2-proxy`) in https://github.com/erkannt/manifests/blob/migration/chart/values.yaml#L31.

**Comment by erkannt** [comment]
No worries. Happy to continue working on it once you've merged this branch.

Feel free to raise those points as issues and I'll try to make PRs where I can.

**Comment by nicolas-geniteau** [comment]
This work looks really great. Any idea when it will be merged ?

**Comment by airhorns** [comment]
Would be really great to get this in!

**Comment by pierluigilenoci** [comment]
@JoelSpeed could you please take a look?

**Comment by AndreaGiardini** [comment]
Can we please get this first version merged and released? After that contributing will become much easier

**Review by JoelSpeed — APPROVED** [review]
I've updated the base branch so we can merge to master, I also updated the title and description a little. I think it's best now if we just get this merged to master, see what works, see what doesn't and bring in fixes as new PRs.

Thanks for your work on this project @erkannt!!!

**Comment by pierluigilenoci** [comment]
@JoelSpeed great news! Would it also be possible to have an update of the chart readme quickly, at least for the installation instructions? The guide still has the reference to stable. Thank you again.

**Pull request merged**
